examples/add_machine.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3.5
+
+"""
+This example:
+
+1. Connects to the current model
+2. Creates two machines and a lxd container
+3. Deploys charm to the lxd container
+
+"""
+import logging
+
+from juju import loop
+from juju.model import Model
+
+MB = 1024 * 1024
+GB = MB * 1024
```

Comment:
If you take a look in constraints.py, these conversions are already built, and they use megabytes, rather than bytes, as a unit, per the Go code on the core side. I'm fairly certain that these numbers will give you machines with much more memory than expected (unless I'm missing a line where you downsize stuff before passing it to the API.)